pyqtgraph/graphicsItems/FillBetweenItem.py

```diff
@@ -69,14 +68,19 @@ def updatePath(self):
 
         path = QtGui.QPainterPath()
         transform = QtGui.QTransform()
+
         ps1 = paths[0].toSubpathPolygons(transform)
         ps2 = paths[1].toReversed().toSubpathPolygons(transform)
         ps2.reverse()
+
         if len(ps1) == 0 or len(ps2) == 0:
             self.setPath(QtGui.QPainterPath())
             return
         
-            
         for p1, p2 in zip(ps1, ps2):
-            path.addPolygon(p1 + p2)
+            intersection = p1.intersected(p2)
+            if not intersection.isEmpty():
+                for i in range(intersection.size()):
+                    path.lineTo(intersection.at(i))
```

Comment:
moveTo and lineTo are problematic due to numerical precision issues in those methods (if memory serves me right they don't work when dealing with values < 10^13. If feasible would be better to create a new QPainterPath (even if it's a line between to points) and join it to the existing one.  An example of this kind of usage can be found in ErrorBarItem.

Reply:
Thank you for your help, we will correct this.

Reply:
Hi @BousquetSophie,

I wanted to get your PR merged into the next release which I plan on making shortly, so I added the change I wanted to your PR, I hope you don't mind!

Reply:
There are no worries.